File: linkedlist/Q83.java
/**
 * https://leetcode.com/problems/remove-duplicates-from-sorted-list/
 */
public class Q83 {
    public ListNode deleteDuplicates(ListNode head) {
        if (head == null || head.next == null) return head;

        ListNode dummy = new ListNode(head.val - 1, head), ptr = dummy;
        int prev = dummy.val;
        while (ptr.next != null) {
            if (ptr.next.val == prev) {
                ptr.next = ptr.next.next;
            } else {
                prev = ptr.next.val;
                ptr = ptr.next;
            }
        }

        return dummy.next;
    }
}

File: LinkedList/Q83.py
# https://leetcode.com/problems/remove-duplicates-from-sorted-list/

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        if not head:
            return None

        ptr = dummy = ListNode(head.val - 1, head)
        prevVal = ptr.val
        while ptr.next:
            if ptr.next.val == prevVal:
                ptr.next = ptr.next.next
            else:
                ptr = ptr.next
                prevVal = ptr.val
        return dummy.next
